src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.ExtensionMethodSymbolComputer.cs

```diff
@@ -0,0 +1,539 @@
+                var assembly = compilation.Assembly;
+                var internalsVisible = currentAssembly.IsSameAssemblyOrHasFriendAccessTo(assembly);
+
+                var matchingMethodSymbols = GetPotentialMatchingSymbolsFromAssembly(
+                    compilation.Assembly, filter, internalsVisible, cancellationToken);
+
+                return isCurrentProject
+                    ? GetExtensionMethodsForSymbolsFromSameCompilation(
+                        matchingMethodSymbols, cancellationToken)
+                    : GetExtensionMethodsForSymbolsFromDifferentCompilation(
+                        matchingMethodSymbols, cancellationToken);
+            }
+
+            private async Task<ImmutableArray<IMethodSymbol>?> GetExtensionMethodSymbolsFromPeReferenceAsync(
+                PortableExecutableReference peReference,
+                bool forceIndexCreation,
+                CancellationToken cancellationToken)
+            {
+                var index = await SymbolTreeInfo.GetInfoForMetadataReferenceAsync(
+                    Solution, peReference, loadOnly: !forceIndexCreation, cancellationToken).ConfigureAwait(false);
+
+                if (index == null)
+                {
+                    // Returns null to indicate index not ready
+                    return null;
+                }
+
+                if (index.ContainsExtensionMethod && SemanticModel.Compilation.GetAssemblyOrModuleSymbol(peReference) is IAssemblySymbol assembly)
+                {
+                    var filter = CreateAggregatedFilter(index);
+                    var internalsVisible = SemanticModel.Compilation.Assembly.IsSameAssemblyOrHasFriendAccessTo(assembly);
+
+                    var matchingMethodSymbols = GetPotentialMatchingSymbolsFromAssembly(
+                        assembly, filter, internalsVisible, cancellationToken);
+
+                    return GetExtensionMethodsForSymbolsFromSameCompilation(
+                        matchingMethodSymbols, cancellationToken);
+                }
+
+                return ImmutableArray<IMethodSymbol>.Empty;
+            }
+
+            private ImmutableArray<IMethodSymbol> GetExtensionMethodsForSymbolsFromDifferentCompilation(
+                MultiDictionary<ITypeSymbol, IMethodSymbol> matchingMethodSymbols,
+                CancellationToken cancellationToken)
+            {
+                using var _ = ArrayBuilder<IMethodSymbol>.GetInstance(out var builder);
+
+                // Matching extension method symbols are grouped based on their receiver type.
+                foreach (var (declaredReceiverType, methodSymbols) in matchingMethodSymbols)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    var declaredReceiverTypeInCurrentCompilation = SymbolFinder.FindSimilarSymbols(declaredReceiverType, SemanticModel.Compilation).FirstOrDefault();
```

Comment:
you might want to doc why you need to find similar symbols.  i assume part of it is for cross lang scenarios (do you tesst those btw?)

Reply:
CurrentCopmilation sounds a bit weird.  you mean the originating-compilation where the user is at?